cli/azd/pkg/infra/azure_resource_manager.go

```diff
@@ -38,17 +38,38 @@ func NewAzureResourceManager(azCli azcli.AzCli) *AzureResourceManager {
 	}
 }
 
+var hasTargetResourceData = func(operation *armresources.DeploymentOperation) bool {
+	return operation.Properties.TargetResource != nil
+}
+
+var isDeploymentType = func(operation *armresources.DeploymentOperation) bool {
+	return *operation.Properties.TargetResource.ResourceType == string(AzureResourceTypeDeployment)
+}
+
+var hasResourceType = func(operation *armresources.DeploymentOperation) bool {
```

Comment:
What pushed you to follow this pattern for defining these helpers instead of using the form of `func` where you declared a named function (i.e. why is this not `func hasResourceType(...) {`) given that we are declaring these functions at package level?

Reply:
I think this was a copy/paste side-effect... Because initially, this was created inside the method which was calling it...  and when I realized it was required from another method, I just moved it to be package-scoped for any method and forgot to change the syntax.  I'll change it